### ISSUE TITLE
Incorrect Autoimport paths

### ISSUE THREAD

**Issue by SchumacherFM** [go-lang-plugin-org/go-lang-idea-plugin#2056]
Kudos for all the awesome work which you are doing! I really love it!

The following is a quite annoying bug which hunts me for some months.

TL;DR: Your Intellij plugin is doing the autoimport of packages differently than goimports from Brad Fitzpatrick.

I'm on OSX with Intellij 15 and self compiled plugin with the latest Github commit from branch master.

Following example: When package errors is not yet included in a file and I type somewhere in the code `errors.Ne...` then the drown down occurs and suggest some packages and functions. AFAIK it suggest the longest name and when you're too fast and press enter that package gets imported.

In my GOPATH I have many checked out repos and when there are duplicated package names with the standard library your plugin importer takes that longest name to import whereas goimports takes the correct name of standard lib package. The printscreen shows an example:

![autoimportdiff](https://cloud.githubusercontent.com/assets/372027/11500387/28b7ac12-982d-11e5-9de2-d9a678bba2d6.png)

Hope that this information is enough for you to fix it :-)

**Comment by dlsniper** [comment]
I think there was a suggestion to check what the other packages / files in the package import and use that as the smart import.

**Comment by SchumacherFM** [comment]
what if there are no other files in my package?
both errors packages have `errors.New()` so I would like to suggest to prefer the standard lib import rather than some other weird path 8-)

**Comment by zolotov** [comment]
> Following example: When package errors is not yet included in a file and I type somewhere in the code errors.Ne... then the drown down occurs and suggest some packages and functions. 

It's not autoimport and it cannot be matched with `goimports` logic in any way.

> AFAIK it suggest the longest name and when you're too fast and press enter that package gets imported.

No, the order depends only on usage statistic.

> both errors packages have errors.New() so I would like to suggest to prefer the standard lib import rather than some other weird path 8-)

Does #1900 will help you with the issue?

**Comment by SchumacherFM** [comment]
Thank you the for the explanations!

That would help :-) Once #1900 is implemented we can close this or even now.

**Comment by SchumacherFM** [comment]
Usage statistic globally or in the current repo I'm working on?

Can the usage statistic be defined more granular?

**Comment by zolotov** [comment]
> That would help :-) Once #1900 is implemented we can close this or even now.

Thanks for answer, will close it after #1990 

> Usage statistic globally or in the current repo I'm working on?

As far as I know the statistic is about selecting particular item from completion list, it's counted on application level. It's platform feature and we cannot modify it from plugin.

**Comment by SchumacherFM** [comment]
stale - won't fix or already fixed in Goland
